src/data/equiv/basic.lean

```diff
@@ -838,6 +838,13 @@ def subtype_pi_equiv_pi {α : Sort u} {β : α → Sort v} {p : Πa, β a → Pr
   by { rintro ⟨f, h⟩, refl },
   by { rintro f, funext a, exact subtype.eq' rfl }⟩
 
+def subtype_prod_equiv_prod {α : Type u} {β : Type v} {p : α → Prop} {q : β → Prop} :
```

Comment:
We already have `equiv.set.prod` which is defeq to this one. I'd reuse one of the definitions in the other.

Reply:
Ah, good catch. Do you think it's fine to have both statements? In my use case, `equiv.set.prod` is phrased too differently for me to switch it in cleanly.

Reply:
I suggested to reuse the definition (e.g. `:= equiv.set.prod {a | p a} {b | q b}`), not to drop this one.